docs/source/workingcopy.rst

```diff
@@ -0,0 +1,228 @@
+    "description": "This is a folder with two documents",
+    "review_state": "private",
+    "title": "My Folder"
+  },
+  "previous_item": {
+    "@id": "http://localhost:55001/plone/folder/doc2",
+    "@type": "Document",
+    "description": "",
+    "title": "A document within a folder"
+  },
+  "relatedItems": [],
+  "review_state": "private",
+  "rights": "",
+  "subjects": [],
+  "table_of_contents": null,
+  "text": null,
+  "title": "My Document",
+  "version": "current",
+  "versioning_enabled": true
+  working_copy: {
+    "@id": "http://localhost:55001/plone/working_copy_of_document",
+  }
+}
+
+the GET content of any a working copy also returns the original (``working_copy_of``)::
+
+GET /plone/document HTTP/1.1
+Accept: application/json
+Authorization: Basic YWRtaW46c2VjcmV0
+
+HTTP/1.1 200 OK
+Content-Type: application/json
+
+{
+  "@components": {
+    "actions": {
+      "@id": "http://localhost:55001/plone/working_copy_of_document/@actions"
+    },
+    "breadcrumbs": {
+      "@id": "http://localhost:55001/plone/working_copy_of_document/@breadcrumbs"
+    },
+    "contextnavigation": {
+      "@id": "http://localhost:55001/plone/working_copy_of_document/@contextnavigation"
+    },
+    "navigation": {
+      "@id": "http://localhost:55001/plone/working_copy_of_document/@navigation"
+    },
+    "types": {
+      "@id": "http://localhost:55001/plone/working_copy_of_document/@types"
+    },
+    "workflow": {
+      "@id": "http://localhost:55001/plone/working_copy_of_document/@workflow"
+    }
+  },
+  "@id": "http://localhost:55001/plone/working_copy_of_document",
+  "@type": "Document",
+  "UID": "SomeUUID000000000000000000000005",
+  "allow_discussion": false,
+  "changeNote": "",
+  "contributors": [],
+  "created": "1995-07-31T13:45:00",
+  "creators": [
+    "admin"
+  ],
+  "description": "",
+  "effective": null,
+  "exclude_from_nav": false,
+  "expires": null,
+  "id": "my-document",
+  "is_folderish": false,
+  "language": "",
+  "layout": "document_view",
+  "modified": "1995-07-31T17:30:00",
+  "next_item": {},
+  "parent": {
+    "@id": "http://localhost:55001/plone/folder",
+    "@type": "Folder",
+    "description": "This is a folder with two documents",
+    "review_state": "private",
+    "title": "My Folder"
+  },
+  "previous_item": {
+    "@id": "http://localhost:55001/plone/folder/doc2",
+    "@type": "Document",
+    "description": "",
+    "title": "A document within a folder"
+  },
+  "relatedItems": [],
+  "review_state": "private",
+  "rights": "",
+  "subjects": [],
+  "table_of_contents": null,
+  "text": null,
+  "title": "My Document",
+  "version": "current",
+  "versioning_enabled": true
+  working_copy_of: {
+    "@id": "http://localhost:55001/plone/document",
+  }
+}
+
+Check-in
+---------
+
+Once the user has finished editing the working copy and wants to update the original
+with the changes in there, or "check-in" the working copy.
+
+PATCH /plone/document/@workingcopy HTTP/1.1
```

Comment:
@sneridagh I am unsure about that one. Though, I do not have any obvious better choice. It feels a bit odd to do a PATCH on the working copy when you actually replace (and change) the original document instead. I guess we have to live with that double-meaning of "@workingcopy" that for GET/POST it represents the working copy object and for PATCH it is a working-copy operation. I need to think about this and maybe sleep over it. I am just talking out loud here from the top of my head.

Reply:
The PATCH here is in the original document, so it's the one to be changed. Well, let's see how it goes the implementation.